src/runtime/server.ts

```diff
@@ -49,6 +58,7 @@ export default defineEventHandler(async (event): Promise<any> => {
           ...endpoint.headers,
           ...headers,
         },
+        cookies,
```

Comment:
You are absolutely right! My local test passed because I use the headers config in nuxt.config and thus the header was set even though it was not passed in the server handler. The new commit should fix this issue. Please note that the order of the headers does somewhat matter. If we set the cookie before the endpoint.headers we are able to manually overwrite the cookie in the nuxt.config file (for development / testing purposes)